cognite_toolkit/_cdf_tk/commands/_migrate/canvas.py

```diff
@@ -1,18 +1,145 @@
+        dry_run: bool = False,
+        verbose: bool = False,
+    ) -> None:
+        canvas = client.canvas.industrial.retrieve(external_id=external_id)
+        if canvas is None:
+            self.warn(MediumSeverityWarning(f"Canvas with external ID '{external_id}' not found. Skipping.. "))
+            return
+        update = canvas.as_write()
+        to_migrate = [
+            ref
+            for ref in update.container_references
+            if ref.container_reference_type in self.asset_centric_resource_types
+        ]
+        if not to_migrate:
+            self.warn(
+                LowSeverityWarning(
+                    f"Canvas with name '{canvas.canvas.name}' does not have any asset-centric references. Skipping.. "
+                )
+            )
+        if verbose:
+            self.console(f"Found canvas: {canvas.canvas.name}")
+        reference_ids = [ref.as_asset_centric_id() for ref in to_migrate]
+        instance_sources = client.migration.instance_source.retrieve(reference_ids)
+        source_by_reference_id = {source.as_asset_centric_id(): source for source in instance_sources}
+        missing = set(reference_ids) - set(source_by_reference_id.keys())
+        if missing:
+            self.warn(
+                HighSeverityWarning(
+                    f"Canvas '{canvas.canvas.name}' has references to resources that are not been migrated: {humanize_collection(missing)}. Skipping.. "
+                )
+            )
+            return
+        if dry_run:
+            self.console(
+                f"Canvas '{canvas.canvas.name}' is ready for migration all {len(instance_sources)} references asset-centric resources found."
+            )
+            return
+        if verbose:
+            self.console(
+                f"Migrating canvas '{canvas.canvas.name}' with {len(instance_sources)} references to asset-centric resources."
+            )
+        backup = canvas.as_write().create_backup()
+
+        update.container_references = [
+            ref
+            for ref in update.container_references
+            if ref.container_reference_type not in self.asset_centric_resource_types
+        ]
+        for ref in to_migrate:
+            source = source_by_reference_id[ref.as_asset_centric_id()]
+            fdm_ref = self.migrate_container_reference(ref, source, canvas.canvas.external_id)
+            update.fdm_instance_container_references.append(fdm_ref)
+
+        try:
+            client.canvas.industrial.create(backup)
+        except CogniteException as e:
+            raise ToolkitMigrationError(f"Failed to create backup for canvas '{canvas.canvas.name}': {e!s}. ")
+        try:
+            client.canvas.industrial.update(update)
+        except CogniteException as e:
+            raise ToolkitMigrationError(
+                f"Failed to migrate canvas '{canvas.canvas.name}': {e!s}. A backup was created with external ID '{backup.canvas.external_id}'."
+            )
```

Comment:
It might be beneficial to do `raise ... from e` for these since they're very broad exceptions.

Reply:
Could just log the CogniteException at `debug` level too and enable debug logging if toolkit is executed with `-v`?

Reply:
nice!